src/factory.js

```diff
@@ -62,10 +62,31 @@ BlockFactory.oldDir = null;
 
 /*
  * The starting XML for the Block Factory main workspace. Contains the
- * unmovable, undeletable factory_base block.
- */
-BlockFactory.STARTER_BLOCK_XML_TEXT = '<xml><block type="factory_base" ' +
+ * unmovable, undeletable factory_base block. Allows user input to change XML
+ * starter block.
+ * @param {string} inputType Type of input (statement, value, dummy).
+ * @param {string} blockStarterText Starter text to place on block, given by user (optional).
+ * @param {string} blockTypeName Name of block, given by user.
+*/
+BlockFactory.insertXMLInputs = function(inputType, blockStarterText, blockTypeName) {
+  inputType = inputType || 'input_statement';
+  blockTypeName = blockTypeName || 'my_block';
+  var textXMLStarter = '';
+
+  // Adds optional text to custom block.
+  if(blockStarterText.trim() !== '') {
+    textXMLStarter = '<value name="FIELDS">' +
+    '<block type="field_static">' +
+    '<field name="TEXT">' + blockStarterText + '</field></block></value>';
+  }
+
+  var CUSTOM_XML_STARTER = '<xml><block type="factory_base" ' +
```

Comment:
Since we're moving into ES6 land, this is a really appropriate place for a [multiline template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Multi-line_strings).